kernel/kernel-api/src/main/java/io/delta/kernel/internal/rowtracking/MaterializedRowTrackingColumn.java

```diff
@@ -175,24 +173,28 @@ public static List<StructField> convertToPhysicalColumn(
       throw DeltaErrors.missingRowTrackingColumnRequested(logicalField.getName());
     }
 
-    if (logicalField.getName().equals(METADATA_ROW_ID_COLUMN_NAME)) {
+    if (logicalField.getMetadataColumnType() == MetadataColumn.ROW_ID) {
       List<StructField> physicalFields = new ArrayList<>(2);
       physicalFields.add(
           new StructField(
-              ROW_ID.getPhysicalColumnName(metadata.getConfiguration()),
+              MATERIALIZED_ROW_ID.getPhysicalColumnName(metadata.getConfiguration()),
               LongType.LONG,
               true /* nullable */));
-      if (logicalSchema.indexOf(StructField.METADATA_ROW_INDEX_COLUMN_NAME) == -1) {
-        physicalFields.add(SchemaUtils.createInternalColumn(StructField.METADATA_ROW_INDEX_COLUMN));
+      if (!logicalSchema.contains(MetadataColumn.ROW_INDEX)) {
+        physicalFields.add(
+            SchemaUtils.createInternalColumn(
+                StructField.createMetadataColumn(
+                    StructField.DEFAULT_ROW_INDEX_COLUMN_NAME, MetadataColumn.ROW_INDEX)));
```

Comment:
Maybe we can have a `DEFAULT_ROW_INDEX_STRUCT_FIELD` in MetadataColumn? And use that here + in `ScanImpl` and anywhere else we construct it using the default name?

Reply:
I think I was mis-understanding what was stored in MetadataColumn. Maybe this would belong in StructField instead?

Reply:
I added `DEFAULT_ROW_INDEX_COLUMN` to `StructField.java`.